llama-index-integrations/readers/llama-index-readers-file/llama_index/readers/file/upstage/base.py

```diff
@@ -0,0 +1,207 @@
+
+OutputType = Literal["text", "html"]
+SplitType = Literal["none", "element", "page"]
+
+
+def validate_api_key(api_key: str) -> None:
+    """
+    Validates the provided API key.
+
+    Args:
+        api_key (str): The API key to be validated.
+
+    Raises:
+        ValueError: If the API key is empty or None.
+
+    Returns:
+        None
+    """
+    if not api_key:
+        raise ValueError("API Key is required for Upstage Document Reader")
+
+
+def validate_file_path(file_path: str) -> None:
+    """
+    Validates if a file exists at the given file path.
+
+    Args:
+        file_path (str): The path to the file.
+
+    Raises:
+        FileNotFoundError: If the file does not exist at the given file path.
+    """
+    if not os.path.exists(file_path):
+        raise FileNotFoundError(f"File not found: {file_path}")
+
+
+def parse_output(data: dict, output_type: OutputType) -> str:
+    """
+    Parse the output data based on the specified output type.
+
+    Args:
+        data (dict): The data to be parsed.
+        output_type (str): The type of output to parse.
+
+    Returns:
+        str: The parsed output.
+
+    Raises:
+        ValueError: If the output type is invalid.
+    """
+    if output_type == "text":
+        return data["text"]
+    elif output_type == "html":
+        return data["html"]
+    else:
+        raise ValueError(f"Invalid output type: {output_type}")
+
+
+class UpstageDocumentReader(BaseReader):
+    def __init__(
+        self,
+        api_key: str = None,
+        api_base: str = None,
+    ):
+        """
+        Initializes a new instance of the Base class.
+
+        Args:
+            api_key (str, optional): The API key to be used for authentication. Defaults to an empty string.
+            api_base (str, optional): The base URL for the API. Defaults to LAYOUT_ANALYZER_URL.
+        """
+        self.api_key = get_from_param_or_env(
+            "UPSTAGE_API_KEY", api_key, "UPSTAGE_API_KEY"
+        )
+        self.api_base = get_from_param_or_env(
+            "UPSTAGE_API_BASE", api_base, "UPSTAGE_API_BASE", LAYOUT_ANALYZER_URL
+        )
+
+        validate_api_key(self.api_key)
+
+    def _get_response(self, file_path) -> requests.Response:
+        """
+        Sends a POST request to the specified URL with the document file
+        and returns the response.
+
+        Returns:
+            requests.Response: The response object from the API call.
+
+        Raises:
+            ValueError: If there is an error in the API call.
+        """
+        try:
+            headers = {"Authorization": f"Bearer {self.api_key}"}
+            files = {"document": open(file_path, "rb")}
+            response = requests.post(self.api_base, headers=headers, files=files)
+        except requests.exceptions.RequestException as e:
+            raise ValueError(f"API call error: {e}")
+        finally:
+            files["document"].close()
+
+        return response.json()
+
+    def load(
```

Comment:
I believe this method should be `load_data(file_path, ..., **kwargs)` ? Double check the base class 👁️ https://github.com/run-llama/llama_index/blob/161405793c8827f9d3ff5acc98906d06b408f27d/llama-index-core/llama_index/core/readers/base.py#L35